src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1402,13 +1413,18 @@ public static String verifyRetentionTime(final String retentionTimeInMin) {
     public static String verifyVirtualNetwork(
             final ServicePrincipal servicePrincipal,
             final String virtualNetworkName,
+            final String virtualNetworkResourceGroupName,
             final String subnetName,
             final boolean usePrivateIP,
             final String resourceGroupName) {
         if (StringUtils.isNotBlank(virtualNetworkName)) {
-            Network virtualNetwork = getVirtualNetwork(servicePrincipal, virtualNetworkName, resourceGroupName);
+            String finalResourceGroupName =  resourceGroupName;
+            if (StringUtils.isNotBlank(virtualNetworkResourceGroupName)) {
```

Comment:
From my understanding, we use `virtualNetworkResourceGroupName` if and only if both `virtualNetworkName` and `virtualNetworkResourceGroupName` are not blank. Otherwise we use the default `resourceGroupName`.

Seems both sides are consistent?